pkg/reconciler/instances/istio/reset/pod/handler.go

```diff
@@ -55,10 +63,24 @@ type NoActionHandler struct {
 
 func (i *NoActionHandler) ExecuteAndWaitFor(context context.Context, object CustomObject) error {
 	if i.debug {
-		i.log.Debugf("Not doing any action for: %s/%s/%s", object.Kind, object.Namespace, object.Name)
+		i.log.Debugf("annotating object where we don't perform proxy reset for: %s/%s/%s", object.Kind, object.Namespace, object.Name)
 	}
-
-	return nil
+	err := k8sRetry.RetryOnConflict(k8sRetry.DefaultRetry, func() error {
+		pod, err := i.kubeClient.CoreV1().Pods(object.Namespace).Get(context, object.Name, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		if pod.Annotations == nil {
```

Comment:
`pod.Annotations` is a map underneath. Is there a chance that it could be empty but not nil?